src/tpm2.c

```diff
@@ -5540,10 +5541,20 @@ int TPM2_GetName(TPM2_CTX* ctx, UINT32 handleValue, int handleCnt, int idx, TPM2
 void TPM2_SetupPCRSel(TPML_PCR_SELECTION* pcr, TPM_ALG_ID alg, int pcrIndex)
 {
     if (pcr && pcrIndex >= (int)PCR_FIRST && pcrIndex <= (int)PCR_LAST) {
-        pcr->count = 1;
-        pcr->pcrSelections[0].hash = alg;
-        pcr->pcrSelections[0].sizeofSelect = PCR_SELECT_MIN;
-        pcr->pcrSelections[0].pcrSelect[pcrIndex >> 3] = (1 << (pcrIndex & 0x7));
+        pcr->pcrSelections[pcr->count].hash = alg;
+        pcr->pcrSelections[pcr->count].sizeofSelect = PCR_SELECT_MIN;
+        pcr->pcrSelections[pcr->count].pcrSelect[pcrIndex >> 3] = (1 << (pcrIndex & 0x7));
+        pcr->count++;
```

Comment:
This PR was already open on my fork. Closing in favor of yours. However there is feedback in that PR you did not capture/fix here. For example: https://github.com/wolfSSL/wolfTPM/pull/244#discussion_r1130737943

Reply:
I've updated this function based on HarryR's advice